Modules/IO/RAW/include/itkRawImageIO.hxx

```diff
@@ -206,50 +265,54 @@ void RawImageIO< TPixel, VImageDimension >
-      memcpy((char *)tempBuffer, buffer, numberOfBytes);          \
-      InternalByteSwapperType::SwapRangeFromSystemToBigEndian(          \
-        (StrongType *)tempBuffer, numberOfComponents);                  \
-      file.write((char *)tempBuffer, numberOfBytes);                            \
-      delete[] tempBuffer;                                              \
-      }                                                                 \
-    else                                                                \
-      {                                                                 \
-      file.write(static_cast< const char * >( buffer ), numberOfBytes); \
-      }                                                                 \
-    }
-
     // Swap bytes if necessary
-    if itkWriteRawBytesAfterSwappingMacro(unsigned short, USHORT)
-    else if itkWriteRawBytesAfterSwappingMacro(short, SHORT)
-    else if itkWriteRawBytesAfterSwappingMacro(char, CHAR)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned char, UCHAR)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned int, UINT)
-    else if itkWriteRawBytesAfterSwappingMacro(int, INT)
-    else if itkWriteRawBytesAfterSwappingMacro(long, LONG)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned long, ULONG)
-    else if itkWriteRawBytesAfterSwappingMacro(float, FLOAT)
-    else if itkWriteRawBytesAfterSwappingMacro(double, DOUBLE)
+    if ( this->GetComponentType() == USHORT )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned short >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == SHORT )
+    {
+      itkWriteRawBytesAfterSwappingMacro< short >( buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == CHAR )
+    {
+      itkWriteRawBytesAfterSwappingMacro< char >( buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == UCHAR )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned char >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == UINT )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned int >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == INT )
+    {
+      itkWriteRawBytesAfterSwappingMacro< int >( buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == LONG )
+    {
+      itkWriteRawBytesAfterSwappingMacro< long >( buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == ULONG )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned long >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == FLOAT )
```

Comment:
Again LONGLONG and ULONGLONG are missing. We probably forgot to add them when here we originally introduced them.